net-libs/libwebsockets/libwebsockets-3.0.0.recipe

```diff
@@ -0,0 +1,89 @@
+SUMMARY="C library for lightweight websocket clients and servers"
+DESCRIPTION="Libwebsockets (LWS) is a flexible, lightweight pure C library \
+for implementing modern network protocols easily with a tiny footprint, \
+using a nonblocking event loop."
+HOMEPAGE="https://libwebsockets.org/"
+COPYRIGHT="2010-2018 Andy Green"
+LICENSE="GNU LGPL v2.1"
+REVISION="1"
+SOURCE_URI="https://github.com/warmcat/libwebsockets/archive/v$portVersion.tar.gz"
+CHECKSUM_SHA256="a6b611c212c52f161f70556339fdaa199b7e9b6a167c4638e086d19db75d6290"
+SOURCE_FILENAME="libwebsockets-$portVersion.tar.gz"
+SOURCE_DIR="libwebsockets-$portVersion"
+
+ARCHITECTURES="?x86_gcc2 ?x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+libVersion="$portVersion"
+libVersionCompat="$libVersion compat >= ${libVersion%%.*}"
+
+PROVIDES="
+	libwebsockets$secondaryArchSuffix = $portVersion
+	lib:libwebsockets$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libcrypto$secondaryArchSuffix
+	lib:libssl$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	libwebsockets${secondaryArchSuffix}_devel = $portVersion
+	devel:libwebsockets$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES_devel="
+	libwebsockets$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libcrypto$secondaryArchSuffix
+	devel:libssl$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	"
+
+BUILD_PREREQUIRES="
+	cmd:gcc$secondaryArchSuffix
+	cmd:cmake
+	cmd:make
+	cmd:openssl
+	"
+
+defineDebugInfoPackage libwebsockets$secondaryArchSuffix \
+	$libDir/libwebsockets.so.13
```

Comment:
```diff
-defineDebugInfoPackage libwebsockets$secondaryArchSuffix \
-	$libDir/libwebsockets.so.13
-
```
(The compilation is not being done with `-g`, so it is better to not include a debuginfo package.)

Reply:
BTW, what I did to find out if `-g` was being passed to the compiler was:
1) comment out the call to `defineDebugInfoPackage` in the recipe, then build it;
2) extract the contents of the archive to a temporary directory, or install it;
3) run
```
readelf -S /path/to/libwebsockets.so.13 | grep debug
```
Here we're not getting any output, so this means that if we did call `defineDebugInfoPackage` we would get a debuginfo package but its contents would be quite useless.